src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key:keyof T): GroupsMap<T> {
+  const obj: GroupsMap<T> = {};
+
+  items.forEach((el) => {
+    const val = String(el[key]);
```

Comment:
more preferred syntax
```suggestion
    const val = `${el[key]}`;
```